test/libsolidity/semanticTests/functionCall/call_options_overload.sol

```diff
@@ -3,9 +3,9 @@ contract C {
     function f(uint x, uint y) external payable returns (uint) { return 2; }
     function call() public payable returns (uint v, uint x, uint y, uint z) {
         v = this.f{value: 10}(2);
-        x = this.f{gas: 1000}(2, 3);
-        y = this.f{gas: 1000, value: 10}(2, 3);
-        z = this.f{value: 10, gas: 1000}(2, 3);
+        x = this.f{gas: 10000}(2, 3);
+        y = this.f{gas: 10000, value: 10}(2, 3);
+        z = this.f{value: 10, gas: 10000}(2, 3);
```

Comment:
So I guess this test just always failed on the homestead run without optimizer? And since we reran it with optimizer we didn't notice? Then just increasing the gas value is a good solution! This was probably causing issues due to https://github.com/ethereum/solidity/pull/14511/files#r1302041690, though, right? Otherwise, I'd just have expected an homestead run with ``--accept-updates`` to add the requires-optimizer setting here. Or was there something even weirder happening :-)?

Reply:
> So I guess this test just always failed on the homestead run without optimizer? And since we reran it with optimizer we didn't notice?

I don't think so. This test was and is compiling fine. The failure was only due to a revert and that does not trigger recompilation.

I think it is just runs out of gas. Inconsequential things like removing one of these `this.f()` calls, adding/removing `value` or removing arguments from `f()` make the failure go away too. On the other hand adding a simple storage write in `f()` makes it fail again despite such changes. I'm not sure what exactly causes this, but my bet is on some differences in how the code gets optimized. Maybe the double optimization we used to do was making it fit just under the limit.

Before I gave up debugging it, I got the differences in generated Yul down to this, which does not seem to have anything that would look like an obvious cause (other than potentially too low gas):
```diff
             function fun_call_42() -> var_x_15, var_z_17 {
                 ...

@@ -304,11 +304,14 @@
                 let expr_21_address := convert_t_contract$_C_$47_to_t_address(expr_20_address)
                 let expr_21_functionSelector := 0x13d1aa2e

-                let expr_22 := 0x03e8
+                let expr_22 := 0x0a
+
+                let expr_23 := 0x03e8

                 let expr_24_address := expr_21_address
                 let expr_24_functionSelector := expr_21_functionSelector
-                let expr_24_gas := convert_t_rational_1000_by_1_to_t_uint256(expr_22)
+                let expr_24_value := convert_t_rational_10_by_1_to_t_uint256(expr_22)
+                let expr_24_gas := convert_t_rational_1000_by_1_to_t_uint256(expr_23)

                 let expr_25 := 0x02

@@ -321,7 +324,7 @@
                 mstore(_4, shift_left_224(expr_24_functionSelector))
                 let _5 := abi_encode_tuple_t_rational_2_by_1_t_rational_3_by_1__to_t_uint256_t_uint256__fromStack(add(_4, 4) , expr_25, expr_26)

-                let _6 := call(expr_24_gas, expr_24_address,  0,  _4, sub(_5, _4), _4, 32)
+                let _6 := call(expr_24_gas, expr_24_address,  expr_24_value,  _4, sub(_5, _4), _4, 32)

                 if iszero(_6) { revert_forward_0() }
```

That's on a trimmed down contract with only two calls and two return values in `call()`, but that does not change things much.

Reply:
> This was probably causing issues due to https://github.com/ethereum/solidity/pull/14511/files#r1302041690, though, right?

Unfortunately the link seems broken, since I pushed new code after you posted the comment.

Reply:
Nevermind the link (I meant to link https://github.com/ethereum/solidity/pull/14511#discussion_r1302041690).

The diff above is a bit weird - that looks like a diff between ``this.f{gas: 10000}(2, 3);`` and ``this.f{value: 10, gas: 10000}(2, 3);`` (red passes zero value to the ``call``, green passes ``10``)

Did this fail with or without ``--optimize``?

But yeah, there's probably nothing to worry about here, so maybe looking into it further is mostly wasting time :-).

Reply:
> that looks like a diff between `this.f{gas: 10000}(2, 3);` and `this.f{value: 10, gas: 10000}(2, 3);`

Yes, exactly. I basically left only those two in the function and it was the minimum required to make it fail. Making both calls identical by adding `value` to the first one was resolving the issue. It was indeed weird because I expected changes to `gas` to have such an affect but this was happening when messing with `value`. In general it was hard to pinpoint what exactly triggered it. When I figured out that increasing `gas` by an order of magnitude resolves this, I concluded that it must still be a gas issue and stopped playing with it :)

And yeah, this was compiled `--optimize`.
`solc <file> --ir --evm-version homestead --debug-info none` to be specific.

> But yeah, there's probably nothing to worry about here, so maybe looking into it further is mostly wasting time :-).

That's my impression too. I actually remember solving a similar issue in a similar test failing on homestead in the past. That's what gave me idea to try with increased gas despite the inconsistent behavior. I even checked if it wasn't maybe the same test but apparently not.

Reply:
Ok, let's just leave it at that (and drop homestead soon altogether :-))